Automattic/simplenote-android
# Update Note Widget

### Fix
Update the note widget added in https://github.com/Automattic/simplenote-android/pull/627 with the following:
- add launching configure activity when note not found (i.e. deleted)
- add launching signin activity when not signed in
- add touch feedback on Marshmallow and later
- add analytics tracking
- update configure activity from full-screen to dialog
- update hard-coded values to resources references
- update background color and corner radius
- update views to match app style
- update maximum lines of text
- update loading text and color
- update signin text and color
- update preview image
- update code style
- remove unused import statements
- remove unused resources

The corner radius of the widget was updated from 5dp to 2dp to match the default card view.  The text color and case were updated to differentiate between note content and a note state.  The `simplenote_dark_grey` (`#2D3034`) color and title case are used when the widget is showing note content.  The `simplenote_light_grey` (`#777777`) color and sentence case are used when the widget is in a loading/error state and is not showing content.  A drop shadow was added to the preview image to simulate elevation while placing the widget on the home screen.  The maximum number of lines was updated from 5 to 4 since 5 lines was larger than the widget view and wasn't showing the ellipsis on Android Q.  The configure activity was updated from full-screen to dialog to follow [Material guidelines for widget configure display](https://material.io/design/platform-guidance/android-widget.html).  See the screenshots below for illustration.

![widget_note](https://user-images.githubusercontent.com/3827611/58669803-65087a00-82fa-11e9-9c22-afd985cf84e7.png)

![widget_loading](https://user-images.githubusercontent.com/3827611/58669807-69349780-82fa-11e9-8eac-970809424482.png)

![widget_found](https://user-images.githubusercontent.com/3827611/58669814-6df94b80-82fa-11e9-8022-6b9799165832.png)

![widget_signin](https://user-images.githubusercontent.com/3827611/58669819-72bdff80-82fa-11e9-822c-e2d52d65b6da.png)

![widget_lines](https://user-images.githubusercontent.com/3827611/58669822-76ea1d00-82fa-11e9-84ca-7ed68fc267ed.png)

![widget_preview](https://user-images.githubusercontent.com/3827611/58669825-794c7700-82fa-11e9-975a-ade2a8b351ed.png)

### Test
These steps assume the AOSP launcher (i.e. Nexus/Pixel launcher).  The steps to add the widget to the home screen may be slightly different based on the launcher used.  Other steps should be the same regardless of launcher.
1. Long-press home screen.
2. Tap ***Widgets*** option.
3. Scroll to ***Simplenote*** widget.
4. Notice ***To-Do List*** preview image.
5. Long-press ***Note*** widget.
6. Place ***Note*** widget on home screen.
7. Notice ***Select Note*** dialog with note list.
8. Tap note in list.
9. Notice note title shown in widget.
10. Tap widget.
11. Notice note is shown in app.
12. Go to ***All Notes***.
13. Delete note from Step 8.
14. Go to ***Trash***.
15. Empty trash.
16. Go to home screen.
17. Notice ***Note not fund*** shown in widget.
18. Tap widget.
19. Notice ***Select Note*** dialog with note list.
20. Tap ***Cancel*** button to dismiss dialog.
21. Go to ***Settings***.
22. Tap ***Sign Out*** option.
23. Notice signin screen shown.
24. Go to home screen.
25. Notice ***Sign in to use widget*** shown in widget.
26. Tap widget.
27. Notice signin screen shown.

### Review
@daniloercoli, @maxme, and @nbradbury, I requested you as reviewer since you have contributed to Simplenote in the past.  @bummytime, I requested you as a reviewer to keep you informed of Simplenote changes.  @SylvesterWilmott, I requested you as a reviewer for design.  Only one developer is required to review these changes, but anyone can perform the review.
